src/waypoint.jsx

```diff
@@ -107,9 +110,11 @@ class TargetEventHandlers {
     if (eventHandlers.size === 0) {
       this.target.removeEventListener(
         eventName,
-        this.handleEvent.bind(this, eventName),
+        eventHandlers.handleEvent,
         EVENT_OPTIONS
       );
+
+      eventHandlers.handleEvent = undefined;
```

Comment:
That was unclear... I mean how would you feel about `delete eventHandlers.handleEvent`?